### PR TITLE
allow ip6tables to be disabled

### PR DESCRIPTION
**Pull request by knackaron**
many hardened systems have IPv6 disabled, which
does not allow ip6tables to be running.  allow
ip6tables to be selectively disabled in these cases.

errors when IPv6 is disabled:

Error: Could not start Service[ip6tables]: Execution of '/usr/bin/systemctl start ip6tables' returned 1: Job for ip6tables.service failed because the control process exited with error code. See "systemctl status ip6tables.service" and "journalctl -xe" for details.
Error: /Stage[main]/Firewall::Linux::Redhat/Service[ip6tables]/ensure: change from stopped to running failed: Could not start Service[ip6tables]: Execution of '/usr/bin/systemctl start ip6tables' returned 1: Job for ip6tables.service failed because the control process exited with error code. See "systemctl status ip6tables.service" and "journalctl -xe" for details.

● ip6tables.service - IPv6 firewall with ip6tables
   Loaded: loaded (/usr/lib/systemd/system/ip6tables.service; disabled; vendor preset: disabled)
   Active: failed (Result: exit-code) since Fri 2017-04-07 01:36:45 UTC; 25min ago
  Process: 10257 ExecStart=/usr/libexec/iptables/ip6tables.init start (code=exited, status=1/FAILURE)
 Main PID: 10257 (code=exited, status=1/FAILURE)

Apr 07 01:36:45 el7-1.example.com systemd[1]: Starting IPv6 firewall with ip6tables...
Apr 07 01:36:45 el7-1.example.com ip6tables.init[10257]: ip6tables: Applying firewall rules: ip6tab...r'
Apr 07 01:36:45 el7-1.example.com ip6tables.init[10257]: Error occurred at line: 4
Apr 07 01:36:45 el7-1.example.com ip6tables.init[10257]: Try `ip6tables-restore -h' or 'ip6tables-r...n.
Apr 07 01:36:45 el7-1.example.com ip6tables.init[10257]: [FAILED]
Apr 07 01:36:45 el7-1.example.com systemd[1]: ip6tables.service: main process exited, code=exited,...URE
Apr 07 01:36:45 el7-1.example.com systemd[1]: Failed to start IPv6 firewall with ip6tables.
Apr 07 01:36:45 el7-1.example.com systemd[1]: Unit ip6tables.service entered failed state.
Apr 07 01:36:45 el7-1.example.com systemd[1]: ip6tables.service failed.
Hint: Some lines were ellipsized, use -l to show in full.

### CODE REVIEW

File: manifests/linux/redhat.pp [hunk 1]
```diff
@@ -8,13 +8,24 @@
...
+# [*enable_v6*]
+#   Enable parameter passed onto Service[] resources.
+#   Default: true
+#
+#
 class firewall::linux::redhat (
   $ensure          = running,
+  $ensure_v6       = running,
   $enable          = true,
+  $enable_v6       = true,
```
Comment (hunner) [inline manifests/linux/redhat.pp]: To remain backwards-compatible, `$ensure_v6` and `$enable_v6` have to default to the values of `$ensure` and `$enable`. Unfortunately this means the silly pattern of `undef` and then `pick()` to an underscore value: https://github.com/puppetlabs/puppetlabs-tomcat/blob/master/manifests/instance.pp#L49

**Comment by knackaron** [comment]
@hunner I applied the pick() pattern to init.pp, linux.pp, and linux/redhat.pp, as I did not see any of those classes marked as private, so I want to ensure correct behavior throughout.  Unit tests pass and also validated in a test environment on EL7 hosts.

**Comment by Mears148** [comment]
We definitely need this ability to disable iptablesv6 in our production environment...any timetable as to when a new build will be released with this added functionality?

**Comment by Mears148** [comment]
Bump

**Comment by h0tw1r3** [comment]
Is there something wrong with this? I have been checking this ticket for 2+ months for activity. We're currently using a fork in our environment because of this **bug**.

**Comment by mears148** [comment]
We forked as well using your code Jeff. Thanks!

**Comment by knackaron** [comment]
@hunner Anything holding up this getting merged?  I've had this in production for a few months and it looks like some others have validated the need for it too.

**Comment by tphoney** [comment]
Many thanks for the PR and for your patience. There is always too much going on 😄

**Pull request merged**